### PR TITLE
Fix test in the daily buildbot and make sure to don't loose the current linker.

### PR DESCRIPTION
**Pull request by nouiz**
Also allow the flag mode to be an instance of Mode.

### CODE REVIEW

**Comment by nouiz** [comment]
Note, all the change in this PR except the test split is needed to fix tests in the daily buildbot.

File: theano/gpuarray/tests/test_opt.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from __future__ import absolute_import, print_function, division
-import numpy as np
+from nose.plugins.skip import SkipTest
```
Comment (abergeron) [inline theano/gpuarray/tests/test_opt.py]: SkipTest is already imported below.

**Comment by nouiz** [comment]
fixed

**Comment by nouiz** [comment]
I removed the commits that convert the old Pow fix to the new as it was causing a jenkins failure:

http://earlgrey.iro.umontreal.ca:8080/job/Theano_PR/682/testReport/junit/theano.gpuarray.tests/test_elemwise/test_elemwise_pow/

**Pull request merged**